evalml/data_checks/class_imbalance_data_check.py

```diff
@@ -146,7 +147,19 @@ def validate(self, X, y):
         """
         messages = []
 
+        original_vc = pd.Series(y).value_counts(sort=True)
         y = infer_feature_types(y)
+        new_vc = y.value_counts(sort=True)
+        after_to_before_inference_mapping = {
+            new: old for old, new in zip(original_vc.keys(), new_vc.keys())
+        }
+        before_to_after_inference_mapping = {
+            old: new for new, old in after_to_before_inference_mapping.items()
+        }
+        if str(y.ww.logical_type) not in ["Boolean", "BooleanNullable"]:
+            after_to_before_inference_mapping = {
+                old: old for old in after_to_before_inference_mapping.keys()
+            }
```

Comment:
This logic is a bit convoluted but I think I follow it. Can we move lines 159 - 162 up to before 153 and set it as an if/else for the `after_to_before_inference_mapping`?

Reply:
Sure thing, added some comments for clarity too